posts/2023-09-29_gnomobile/README.md

```diff
@@ -0,0 +1,77 @@
+3. **Toolkit**: We want to make it as easy as possible for devs to start building apps with our framework, so we’ll provide them with install instructions, example apps, and more technical stuff like genproto options to support gRPC and helper functions to parse the render output.
+
+Those should be enough to allow builders to get started on using and experimenting with Gno mobile apps.
+
+- *Support for secure p2p communication, even when the Internet is down?*
+- *Yes, please!*
+
+Something that is not necessarily essential for V1, but for sure will open the doors to some powerful capabilities later on is to add an interface and a constructor to adapt the communication transport. This will make it possible for devs to incorporate other tools like Wesh Network and give their apps the ability to securely and reliably send messages even in very poor network conditions. But that’s a story for another time.
+
+# When will GnoMobile be ready?
+
+V1 is planned for release in mid-December 2023.
+
+Until then, you can check out our progress [here](https://github.com/gnolang/hackerspace/issues/28).
+
+Got feedback or want to drop us a question? Ask away on our [repo](https://github.com/gnolang/gnomobile/issues).
+
+# What does the future look like beyond V1?
+
+We see a lot of potential directions for GnoMobile after the initial release that will improve the user experience, extend its functionality, and make GnoMobile even more secure. We’re still scratching the surface in terms of how far we can take its development, and we look forward to working on further iterations and improvements. Some of our ideas for the future beyond V1 include:
+
+1. Making it easier for developers to **build** **desktop apps** **and** **browser extensions**:
+2. Through GnoMobile, we can gradually enable “desktop” devs to use our React Native gRPC interface to write desktop applications while using existing functionality from the core Go code. This way, developers will not necessarily have to learn Go to leverage its advantages.
+3. Browser extensions are usually written in JavaScript in the same way as in React Native. This opens the door to getting the benefits of Go via the GnoMobile framework. Otherwise, you’d have to either make the Go code run inside the browser extension (which is not easy) or use a remote server (which is not pretty).
+4. Making it possible to **execute smart contracts directly from mobile**.
+
+*Why is this important?*
+
+If you want to add a new message to a blockchain, you need to actually interact with it (the blockchain) and update its state with the new message. However, if you just want to browse through the messages, you can execute the Render function locally without needing to use your network and, at the same time, get the results much faster. This is because the node runs locally on the mobile device without needing to spend crypto coins to get a remote node to do the operation for you.
```

Comment:
With the node running locally, how long do you anticipate a sync to occur, or do you imagine this application to serve more like a light client where it will communicate externally to either a full or archive node?

Reply:
The plan for V1 is a light client. In this section, we discuss future plans to experiment with the full node on the mobile. In this case, you are correct that it is a concern how much time it can require to sync the blockchain onto the mobile device (or if it is possible to only sync part of it). This needs to be investigated.